modules/rte_mis_school/src/Access/SchoolPdfDownloadAccessCheck.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Drupal\rte_mis_school\Access;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Routing\Access\AccessInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\eck\EckEntityInterface;
+
+/**
+ * Determines edit access to the school details mini node.
+ */
+class SchoolPdfDownloadAccessCheck implements AccessInterface {
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The current user.
+   *
+   * @var \Drupal\Core\Session\AccountProxyInterface
+   */
+  protected $currentUser;
+
+  /**
+   * Constructs a SchoolPdfDownloadAccessCheck object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The current user.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, AccountProxyInterface $current_user) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function access(RouteMatchInterface $routeMatch) {
+    // Load the current user.
+    /** @var \Drupal\user\userInterface $curr_user */
+    $curr_user = $this->entityTypeManager->getStorage('user')->load($this->currentUser->id());
+    $curr_user_roles = $curr_user->getRoles();
+    // Get the mini node from the route.
+    $miniNodeId = $routeMatch->getParameters('mini_node')->get('entity_id');
+
+    // Logic for district and block admins.
+    // Load the mini node based on miniNodeId.
+    $currUserLinkedLocation = $curr_user->get('field_location_details')->getString();
+    $miniNode = $this->entityTypeManager->getStorage('mini_node')->loadByProperties([
+      'type' => 'school_Details',
+      'id' => $miniNodeId,
+      'status' => 1,
+    ]);
+    $miniNode = reset($miniNode);
+
+    if ($miniNode instanceof EckEntityInterface) {
+      // Get the location details of mini node.
+      $mini_node_location = $miniNode->get('field_location')->getString();
+      // Load all the parents of the mini node's location id.
+      $locationTermParent = $this->entityTypeManager->getStorage('taxonomy_term')->loadAllParents($mini_node_location);
+      // Create an array of the keys of the locationTermParent.
+      $locationTermParentKeys = array_keys($locationTermParent);
+    }
```

Comment:
This should only be done for block and district admin, not for schools.

Reply:
yes, Modified